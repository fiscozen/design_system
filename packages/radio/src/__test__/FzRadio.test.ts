import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import { FzRadio, FzRadioProps } from "..";

const createWrapper = async (props: FzRadioProps) => {
  const content = mount(FzRadio, {
    props,
  });
  return content;
};

describe("FzRadio", () => {
  it("should render correctly", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should have checked attribute", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      checked: true,
    });

    expect(wrapper.find("input").element.checked).toBe(true);
  });

  it("should have disabled attribute", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      disabled: true,
    });

    expect(wrapper.find("input").element.disabled).toBe(true);
  });

  it("should show error text and error icon", async () => {
    const wrapper = mount(FzRadio, {
      props: {
        label: "Radio",
        size: "md",
        error: true,
      },
      slots: {
        error: "Error text",
      },
    });
    await wrapper.vm.$nextTick();

    expect(wrapper.find("p")).toBeTruthy();
    expect(wrapper.find("p").text()).toBe("Error text");
    expect(wrapper.find("p").find("svg")).toBeTruthy();
  });

  it("should toggle error text when error switch from true to false", async () => {
    const wrapper = mount(FzRadio, {
      props: {
        label: "Radio",
        size: "md",
        error: true,
      },
      slots: {
        error: "Error text",
      },
    });
    await wrapper.vm.$nextTick();

    expect(wrapper.find("p")).toBeTruthy();
    expect(wrapper.find("p").text()).toBe("Error text");

    await wrapper.setProps({ error: false });

    expect(wrapper.find("p").exists()).toBeFalsy();
  });

  it("should not show label when standalone is true", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      standalone: true,
    });

    expect(wrapper.find("label").text()).toBeFalsy();
  });

  it("should have emphasized class", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      emphasis: true,
    });

    expect(wrapper.find("input").classes()).toContain("radio--emphasized");
  });
});
