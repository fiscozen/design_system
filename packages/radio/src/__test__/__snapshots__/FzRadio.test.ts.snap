// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzRadio > should render correctly 1`] = `
"<div class="flex justify-center flex-col w-fit"><input type="radio" id="Radio" class="peer h-0 w-0 absolute radio--medium" tabindex="0" value="Radio"><label for="Radio" class="flex items-start gap-4 h-16 
  before:content-[''] 
  before:inline-block 
  before:border-solid 
  before:border-1 
  before:rounded-full 
  before:ml-4
  peer-checked:before:bg-transparent 
  peer-focus:before:outline 
  peer-focus:before:outline-offset-1 
  peer-focus:before:outline-1 
  peer-focus:before:outline-blue-600 text-md before:h-16 before:w-16 before:mt-4 peer-checked:before:border-[5px] before:border-grey-500"><span class="w-fit">Radio</span></label>
  <!--v-if-->
</div>"
`;
